### PR TITLE
fix(console): refresh UI after API detach from automation source

### PR DESCRIPTION
**Pull request by ankita-gupta21**
## Issue

https://gravitee.atlassian.net/browse/APIM-13877

## Description

Reload General Info via refresh$ instead of re-running ngOnInit, refresh lastApiFetch after detach, and update danger zone detach/read-only state when originContext changes.

## Additional context

**api-general-info:** Replace (reloadDetails)="ngOnInit()" with reloadDetails() that triggers the existing refresh$ subject, so the API is refetched without stacking duplicate subscriptions.
**ApiV2Service.detach():** Chain refreshLastApiFetch() after _detach (same pattern as deploy) so getLastApiFetch() consumers (e.g. API navigation) receive the updated originContext.
**api-general-info-danger-zone:** Move canDetach / isReadOnly into updateOriginContextState() and call it from ngOnChanges when the [api] input changes, so the Detach action hides after reload.

https://github.com/user-attachments/assets/08fc6b75-a86a-4107-b5e7-58f1abf2228d

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/general-info/api-general-info-danger-zone/api-general-info-danger-zone.component.ts [hunk 1]
```diff
@@ -338,6 +338,11 @@ export class ApiGeneralInfoDangerZoneComponent implements OnChanges, OnDestroy,
       .subscribe();
   }
 
+  private updateOriginContextState(): void {
+    this.isReadOnly = this.api.definitionVersion === 'V1' || this.api.originContext?.origin === 'KUBERNETES';
+    this.canDetach = this.api.originContext?.origin === 'KUBERNETES';
+  }
```
[the remark this answers is not in the document]
Comment (ankita-gupta21) [reply gravitee-apim-console-webui/src/management/api/general-info/api-general-info-danger-zone/api-general-info-danger-zone.component.ts:344]: fixed

File: gravitee-apim-console-webui/src/services-ngx/api-v2.service.ts [hunk 2]
```diff
@@ -84,7 +84,9 @@ export class ApiV2Service {
   }
 
   detach(apiId: string): Observable<void> {
-    return this.http.post<void>(`${this.constants.env.v2BaseURL}/apis/${apiId}/_detach`, {});
+    return this.http
+      .post<void>(`${this.constants.env.v2BaseURL}/apis/${apiId}/_detach`, {})
+      .pipe(switchMap(() => this.refreshLastApiFetch()));
   }
```
[the remark this answers is not in the document]
Comment (ankita-gupta21) [reply gravitee-apim-console-webui/src/services-ngx/api-v2.service.ts:95]: fixed

File: gravitee-apim-console-webui/src/management/api/general-info/api-general-info-danger-zone/api-general-info-danger-zone.component.ts [hunk 3]
```diff
@@ -87,8 +87,7 @@ export class ApiGeneralInfoDangerZoneComponent implements OnChanges, OnDestroy,
   ) {}
 
   ngOnInit(): void {
-    this.isReadOnly = this.api.definitionVersion === 'V1' || this.api.originContext?.origin === 'KUBERNETES';
-    this.canDetach = this.api.originContext?.origin === 'KUBERNETES';
+    this.updateOriginContextState();
     this.license$ = this.licenseService.getLicense$();
     this.isOEM$ = this.licenseService.isOEM$();
```
[the remark this answers is not in the document]
Comment (ankita-gupta21) [reply gravitee-apim-console-webui/src/management/api/general-info/api-general-info-danger-zone/api-general-info-danger-zone.component.ts:92]: fixed

**Review by prateektiwari-gravitee — APPROVED** [review]
LGTM
